src/sync/common.py

```diff
@@ -1,29 +1,20 @@
 """Shared methods between both sync scripts."""
 
-from dune_client.file.interface import FileIO
-
 from src.logger import set_log
+from src.models.tables import SyncTable
+from src.post.aws import AWSClient
 
 log = set_log(__name__)
 
 
-def last_sync_block(
-    file_manager: FileIO, last_block_file: str, column: str, genesis_block: int = 0
-) -> int:
-    """Attempts to get last sync block from file, otherwise uses genesis"""
+def last_sync_block(aws: AWSClient, table: SyncTable, genesis_block: int = 0) -> int:
```

Comment:
Could the genesis block here be the earliest block any of our settlement contracts have been created? Just asking in case this could reduce sync time if we don't find any files in the bucket. Under normal circumstances this should probably never happen but maybe that's useful for debugging locally? 
Honestly don't know so I just wanted to bring it up in case you haven't thought about that, yet. 🤷‍♂️

Reply:
We pass genesis blocks in from the script (they are hard coded as the first blocks where data exists) but I could also use the deployment block of our contract as well to be more uniform. However the order rewards didn't start until much later.